### PR TITLE
Refactoring of main file. EA config generating and config reading are taken out in corresponding  files.

### PR DESCRIPTION
**Pull request by Kuyantsev-VP**

### CODE REVIEW

**Comment by Kuyantsev-VP** [comment]
I will try to add all required tests today. Or at least till next meeting.

File: generate_configs.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from jinja2 import Environment, FileSystemLoader
+from config import get_env_vars_by_prefix
+
+
+def get_env(templates_dir):
+    return Environment(loader=FileSystemLoader(templates_dir))
+
+
+class Renderer(object):
```
Comment (rvadim) [inline generate_configs.py]: `(object)` not needed

File: generate_configs.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        self.env = get_env(self.templates_dir)
+
+    def _generate_config(self, template_name, context):
+        template = self.env.get_template(template_name)
+        return template.render(context)
+
+    def generate_ea_config(self, adm_context):
+        return self._generate_config('ea_config.yaml.j2', adm_context)
+
+    def generate_ea_rules(self, usr_configs):
```
Comment (rvadim) [inline generate_configs.py]: Don't use reduction when it's not needed

File: templates/ea_rule.yaml.j2 [hunk 3]
```diff
@@ -14,7 +14,17 @@ timeframe:
 num_events: {{num_events}}
 {% endif %}
 
-filter: {{ filter }}
+{% if filter is defined %}
```
Comment (rvadim) [inline templates/ea_rule.yaml.j2:17]: Why not use just  {{ filter }}?

File: readers.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+from config import validate_user_rule
+
+log = logging.getLogger("__main__")
+
+
+class BaseUsrConfigReader(object):
+    """
+    Base class for user config readers with common interface
+    """
+    def __init__(self):
```
Comment (rvadim) [inline readers.py]: omit this constructor

File: readers.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+from config import get_env_vars_by_prefix
+from config import parse_config
+from config import validate
+from config import validate_user_rule
+
+log = logging.getLogger("__main__")
+
+
+class BaseUsrConfigReader(object):
```
Comment (rvadim) [inline readers.py]: BaseConfigReader

**Review by rvadim — CHANGES_REQUESTED** [review]
Need add tests

File: templates/ea_rule.yaml.j2 (hunk 3 above)
Comment (Kuyantsev-VP) [reply templates/ea_rule.yaml.j2:17]: In general case this property is optional.

Comment (rvadim) [reply templates/ea_rule.yaml.j2:17]:
```jinja2
{% if filter is defined %}
filter: {{ filter | to_yaml }}
{% endif %}

**Review by rvadim — DISMISSED** [review]
LGTM

**Review by dkondyrev — DISMISSED** [review]

**Review by dkondyrev — APPROVED** [review]

**Review by rvadim — APPROVED** [review]
LGTM

**Pull request merged**